helix-lsp/src/lib.rs

```diff
@@ -247,13 +248,56 @@ pub mod util {
         Some(Range::new(start, end))
     }
 
+    fn completion_pos(
+        text: RopeSlice,
+        start_offset: Option<i128>,
+        end_offset: Option<i128>,
+        cursor: usize,
+    ) -> Option<(usize, usize)> {
+        let replacement_start = match start_offset {
+            Some(start_offset) => {
+                let start_offset = cursor as i128 + start_offset;
+                if start_offset < 0 {
+                    return None;
+                }
+                start_offset as usize
+            }
+            None => {
+                cursor
+                    + text
+                        .chars_at(cursor)
+                        .skip(1)
+                        .take_while(|ch| chars::char_is_word(*ch))
+                        .count()
+            }
+        };
+        let replacement_end = match end_offset {
+            Some(end_offset) => {
+                let replacement_end = cursor as i128 + end_offset;
+                if replacement_end > text.len_chars() as i128 {
+                    return None;
+                }
+                replacement_end as usize
+            }
+            None => {
+                cursor
+                    - text
+                        .chars_at(cursor)
+                        .reversed()
+                        .take_while(|ch| chars::char_is_word(*ch))
+                        .count()
+            }
+        };
+        Some((replacement_start, replacement_end))
+    }
+
     /// Creates a [Transaction] from the [lsp::TextEdit] in a completion response.
     /// The transaction applies the edit to all cursors.
     pub fn generate_transaction_from_completion_edit(
         doc: &Rope,
         selection: &Selection,
-        start_offset: i128,
-        end_offset: i128,
+        start_offset: Option<i128>,
+        end_offset: Option<i128>,
```

Comment:
It should be an Option of tuple? There probably won't be a case where only one is present but not the other

Reply:
Yeah that was mostly a lazy change, guess I could make it a tuple 👍